tests/network/bgp/conftest.py

```diff
@@ -0,0 +1,179 @@
+from tests.network.libs.label_selector import LabelSelector
+from utilities.infra import cache_admin_client, get_node_selector_dict
+
+APP_CUDN_LABEL: Final[dict] = {"app": "cudn"}
+BGP_DATA_PATH: Final[Path] = Path(__file__).resolve().parent / "data" / "frr-config"
+CUDN_BGP_LABEL: Final[dict] = {"cudn-bgp": "blue"}
+CUDN_SUBNET_IPV4: Final[str] = "192.168.10.0/24"
+EXTERNAL_SUBNET_IPV4: Final[str] = "172.100.0.0/16"
+
+
+@pytest.fixture(scope="module")
+def vlan_nncp(vlan_base_iface: str, worker_node1: Node) -> Generator[libnncp.NodeNetworkConfigurationPolicy]:
+    """Creates a NodeNetworkConfigurationPolicy with a VLAN interface on the specified cluster's node."""
+    with libnncp.NodeNetworkConfigurationPolicy(
+        name="test-vlan-nncp",
+        desired_state=libnncp.DesiredState(
+            interfaces=[
+                libnncp.Interface(
+                    name=f"{vlan_base_iface}.{os.environ['VLAN_TAG']}",
+                    state=libnncp.NodeNetworkConfigurationPolicy.Interface.State.UP,
+                    type="vlan",
+                    vlan=libnncp.Vlan(id=int(os.environ["VLAN_TAG"]), base_iface=vlan_base_iface),
+                )
+            ]
+        ),
+        node_selector=get_node_selector_dict(node_selector=worker_node1.hostname),
+    ) as nncp:
+        nncp.wait_for_status_success()
+        yield nncp
+
+
+@pytest.fixture(scope="session")
+def br_ex_network_info(worker_node1: Node) -> BrExNetworkInfo:
+    """Retrieves cluster node network information."""
+    return lookup_br_ex_network_info(node_name=worker_node1.name)
+
+
+@pytest.fixture(scope="module")
+def macvlan_nad(
+    vlan_nncp: libnncp.NodeNetworkConfigurationPolicy,
+    cnv_tests_utilities_namespace: Namespace,
+    br_ex_network_info: BrExNetworkInfo,
+) -> Generator[libnad.NetworkAttachmentDefinition]:
+    macvlan_config = libnad.CNIPluginMacvlanConfig(
+        master=vlan_nncp.instance.spec.desiredState.interfaces[0].name,
+        ipam=libnad.Ipam(
+            type="host-local",
+            subnet=br_ex_network_info.cidr_v4,
+            rangeStart=os.environ["EXTERNAL_FRR_STATIC_IPV4"],
+            rangeEnd=os.environ["EXTERNAL_FRR_STATIC_IPV4"],
+            gateway=br_ex_network_info.gateway_v4,
+            routes=[libnad.IpamRoute(dst=DEFAULT_ROUTE_DEST_V4, gw=br_ex_network_info.gateway_v4)],
+        ),
+    )
+
+    with libnad.NetworkAttachmentDefinition(
+        name="macvlan-nad-bgp",
+        namespace=cnv_tests_utilities_namespace.name,
+        config=libnad.NetConfig(name="macvlan-nad-bgp", plugins=[macvlan_config]),
+        client=cache_admin_client(),
+    ) as nad:
+        yield nad
+
+
+@pytest.fixture(scope="module")
+def frr_configmap(workers: list[Node], cnv_tests_utilities_namespace: Namespace) -> Generator[ConfigMap]:
+    """Generates a ConfigMap containing the config files for the external FRR."""
+    frr_config_file_path = BGP_DATA_PATH / "frr.conf"
+    generate_frr_conf(
```

Comment:
The file is created in order to read from it to fill up the CM? Why do we need the OS filesystem to do this? It is very odd. To clarify, if the file is needed as a temp media, there are memory based solution which can do the same.